### DISCUSSION TITLE
Output the wall time taken for each nonlinear iteration

### DISCUSSION THREAD

**Discussion by xueyang94** [idaholab/moose#21454] [Q&A General] [answered]
Is there a way to output to the terminal the wall time taken for each nonlinear iteration?

**Comment by GiudGiud** [comment, marked as answer]
Hello

Not that I know of directly. 
But you can use the perf graph to do that:
https://mooseframework.inl.gov/source/postprocessors/PerfGraphData.html
Just set the section to be the nonlinear solve. 
And execute and output this pretty often (on NONLINEAR).

Guillaume

**Reply by xueyang94** [reply]
How to output the time often? I saw that the execute_on parameter has a "CUSTOM" option. Can we customize it to output on each nonlinear iteration? Also, by executing on TIMESTEP_END, each time step outputs a two-column table. What do the two columns mean?

**Reply by GiudGiud** [reply]
the CUSTOM option is going to be some significant C++ coding. You have to create the new flag, but also add all the entry points for this CUSTOM option. 

NONLINEAR is alreacdy an execute_on option. 

I imagine the two columns are "time in the simulation" and "output of the postprcoessor"
But if you past the output here I can check

**Reply by xueyang94** [reply]
Below is the terminal output at Time Step 6.

```
Time Step 6, time = 0.003, dt = 0.0005

Performing automatic scaling calculation

 0 Nonlinear |R| = 1.373773e+02
      0 Linear |R| = 1.373773e+02
      1 Linear |R| = 1.875720e-13
 1 Nonlinear |R| = 1.737294e-01
      0 Linear |R| = 1.737294e-01
      1 Linear |R| = 6.744421e-16
 2 Nonlinear |R| = 1.970520e-05
      0 Linear |R| = 1.970520e-05
      1 Linear |R| = 3.030205e-17
 3 Nonlinear |R| = 1.460017e-06
      0 Linear |R| = 1.460017e-06
      1 Linear |R| = 2.374850e-18
 4 Nonlinear |R| = 9.426885e-09
 Solve Converged!

Outlier Variable Residual Norms:
  b: 9.426722e-09

Postprocessor Values:
+----------------+----------------+
| time           | time           |
+----------------+----------------+
|   0.000000e+00 |   0.000000e+00 |
|   5.000000e-04 |   1.883826e+00 |
|   1.000000e-03 |   2.369580e+00 |
|   1.500000e-03 |   3.573183e+00 |
|   2.000000e-03 |   4.388828e+00 |
|   2.500000e-03 |   4.634377e+00 |
|   3.000000e-03 |   4.940463e+00 |
+----------------+----------------+
```

**Reply by GiudGiud** [reply]
yup first is time in the simulation and second is time in the real world

**Reply by xueyang94** [reply]
Okay. I have execute_on="NONLINEAR", so I expect the wall time will show after each nonlinear iteration. Why is the time output still at the end of each time step?

**Reply by GiudGiud** [reply]
the execute on of the postprocessor isnt the same as the console output

I m not even sure we can show you the postrpcoessor that often. But we can print it to a CSV file on nonlinear
